components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/internal/dao/impl/ActionManagementDAOImpl.java

```diff
@@ -333,14 +342,20 @@ private void updateEndpoint(ActionDTO updatingActionDTO, ActionDTO existingActio
         if (updatingEndpoint == null) {
             return;
         }
-
         try {
             if (updatingEndpoint.getUri() != null) {
                 updateActionPropertiesInDB(updatingActionDTO.getId(),
                         Collections.singletonMap(URI_PROPERTY, new ActionProperty.BuilderForDAO(
                                 updatingEndpoint.getUri()).build()), tenantId);
             }
-
+            if (updatingEndpoint.getAllowedHeaders() != null) {
+                updateActionPropertyListInDB(updatingActionDTO.getId(), ALLOWED_HEADERS_PROPERTY,
+                        updatingEndpoint.getAllowedHeaders(), tenantId);
+            }
+            if (updatingEndpoint.getAllowedParameters() != null) {
+                updateActionPropertyListInDB(updatingActionDTO.getId(), ALLOWED_PARAMETERS_PROPERTY,
+                        updatingEndpoint.getAllowedParameters(), tenantId);
+            }
```

Comment:
IMO, these properties can be collected if those needs to be updated and can be updated once as a batch using updateActionPropertiesInDB function.

The current approach does multiple db calls to update these properties. Isn't it ?